include/psa/crypto_values.h

```diff
@@ -1111,10 +1145,25 @@
  *                      for the specified AEAD algorithm.
  */
 #define PSA_ALG_AEAD_WITH_SHORTENED_TAG(aead_alg, tag_length)           \
-    (((aead_alg) & ~PSA_ALG_AEAD_TAG_LENGTH_MASK) |                     \
+    (((aead_alg) & ~(PSA_ALG_AEAD_TAG_LENGTH_MASK | PSA_ALG_AEAD_MINIMUM_LENGTH_FLAG)) | \
      ((tag_length) << PSA_AEAD_TAG_LENGTH_OFFSET &                      \
       PSA_ALG_AEAD_TAG_LENGTH_MASK))
 
+/** Retrieve the tag length of a specified AEAD algorithm
+ *
+ * \param aead_alg      An AEAD algorithm identifier (value of type
+ *                      #psa_algorithm_t such that #PSA_ALG_IS_AEAD(\p alg)
+ *                      is true).
+ *
+ * \return              The tag length specified by the input algorithm.
+ * \return              Unspecified if \p alg is not a supported
+ *                      AEAD algorithm or if \p tag_length is not valid
```

Comment:
The final part of the return description should not be used here: " or if \p tag_length is not valid for the specified AEAD algorithm"

Reply:
Note: this will not form part of the PSA Crypto API itself, but it is part of the mbedtls implementation.

Reply:
This might need further thought (@gilles-peskine-arm):
1. This macro is now identical to `PSA_AEAD_TAG_LENGTH` in **crypto_sizes.h**
2. However, the definition of `PSA_AEAD_TAG_LENGTH()` in the v1.0.0 specification has extra parameters that are not yet in the mbedtls implementation - most other v1.0.0 changes to these macros have been merged.

Reply:
Thanks Andrew, I filed this as https://github.com/ARMmbed/mbedtls/issues/4163.

Reply:
As we've discussed privately with @athoelke, it's unworkable to accept arbitrary tag lengths here. There isn't enough room to encode the information. At best, we could require that the macro returns 0 on an invalid tag length, but this requires the macro to 
know precisely which tag lengths are valid for each algorithms. The current wording allows an implementation that does `base_mask | tag_length << SHIFT` if `tag_length` fits within a certain number of bits and 0 otherwise, so tag lengths that are invalid because they are too large are effectively rejected at this point whereas tag lengths that are in a plausible range but not valid for the particular algorithm are rejected at operation time.

We're still discussing how to best limit the extent of unspecified behavior while remaining implementable. In the meantime, let's keep the existing wording.

Reply:
ok fine by me.

Reply:
Done.

Reply:
Unless I am vyer much mistaken, there no identifier `tag_length` that is defined and in scope in the context of this macro: `PSA_ALG_AEAD_GET_TAG_LENGTH(aead_alg)`.

In which case, my original request for changes should stand.

Reply:
Oh! My bad, I got my wires crossed between the documentation of `PSA_ALG_AEAD_WITH_SHORTENED_TAG` and `PSA_ALG_AEAD_GET_TAG_LENGTH`. I'll make a pull request to fix this.

Reply:
@gilles-peskine-arm I think you are not the only one that got your wires crossed here. Thanks @athoelke  for double checking.